gatsby-node.esm.js

```diff
@@ -273,47 +181,30 @@ exports.onCreateNode = async ({ node, getNode, actions }) => {
   }
 };
 
-exports.sourceNodes = async ({ actions, getNodes }) => {
-  const { createNodeField } = actions;
-  const nodes = await getNodes();
-  const contestStatusData = await getContestData();
-
-  nodes.forEach((node, index) => {
-    if (node.internal.type === `ContestsCsv`) {
-      const dataForCurrentContest = contestStatusData.filter(
-        (element) => element.contestId === node.contestid
-      );
-      createNodeField({
-        node,
-        name: `status`,
-        value:
-          dataForCurrentContest.length > 0
-            ? dataForCurrentContest[0].status
-            : undefined,
-      });
-      createNodeField({
-        node,
-        name: `codeAccess`,
-        value:
-          dataForCurrentContest.length > 0
-            ? dataForCurrentContest[0].codeAccess
-            : undefined,
-      });
-      createNodeField({
-        node,
-        name: `type`,
-        value:
-          dataForCurrentContest.length > 0
-            ? dataForCurrentContest[0].type
-            : undefined,
-      });
-    }
+exports.sourceNodes = async ({
+  actions,
+  createContentDigest,
+  createNodeId
+}) => {
+  const { createNode } = actions;
+  const apiContestsData = await getApiContestData();
+  apiContestsData.forEach((contest) => {
+    createNode({
+      ...contest,
+      id: createNodeId(`ContestsCsv-${contest.contestid}`),
+      parent: null,
+      children: [],
+      internal: {
+        type: "ContestsCsv",
+        contentDigest: createContentDigest(contest),
+      },
+    });
```

Comment:
Do we want to create nodes here or do we want to make the API calls in the relevant pages ? as we did for the leaderboard for example?